### PR TITLE
Add support for including graphviz files

### LINKED ISSUE #143: Support including graphviz files

### PR DESCRIPTION
**Pull request by oberien**
Fix #143

### CODE REVIEW

File: src/config/validate.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+use super::{CliArgs, FileOrStdio, FileConfig};
+
+pub trait ValidatePath {
+    /// Validates and canonicalizes this PathBuf to point to a valid folder
+    fn validate_dir(&mut self, error_str: &str);
+    /// Validate and canonicalize this input, checking that the file exists
+    fn validate_file_exists(&mut self, error_str: &str);
+    /// Validate and canonicalize this input, ignoring if the file exists or not
+    fn validate_file_ignore(&mut self, error_str: &str);
+}
```
Comment (197g) [inline src/config/validate.rs]: These methods are panicking, not returning Results. Renaming them to `expect_*` would be clearer.

File: src/config/validate.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+//             output_type, document_type, bibliography, citationstyle, citestyle,
+//             bibstyle, figures, template, lang, fontsize,
+//             oneside, classoptions, titlepage, title, subtitle,
+//             titlehead, author, email, date, publisher,
+//             advisor, supervisor, logo_university, logo_faculty, university,
+//             faculty, thesis_type, location, disclaimer, abstract1,
+//             abstract2, sectionframes, beamertheme, header_includes, geometry
+//         } = self;
+//     }
+// }
```
Comment (197g) [inline src/config/validate.rs]: What's this about?

File: src/generator/iter.rs [hunk 3]
```diff
@@ -193,6 +194,27 @@ impl<'a> Iter<'a> {
...
+                })?;
+                let tag = Tag::Graphviz(Graphviz {
+                    label,
+                    caption,
+                    scale,
+                    width,
+                    height,
+                });
+                self.peek.push_back((WithRange(Event::Text(content.into()), range), self.last_kind));
+                self.peek.push_back((WithRange(Event::End(tag.clone()), range), self.last_kind));
```
Comment (197g) [inline src/generator/iter.rs:215]: Heh. Cute hack. It's not even proper to really say this is a hack. Does this conflict or overlap with the ability to do multiple peeks?

File: src/config/validate.rs (hunk 1 above)
Comment (oberien) [reply src/config/validate.rs]: Whoops, that file shouldn't have been added :D

File: src/generator/iter.rs (hunk 3 above)
Comment (oberien) [reply src/generator/iter.rs:215]: It could enable multiple peeks. For now, the `peek` method still only peeks one element.

Comment (197g) [reply src/generator/iter.rs:215]: I recall some instance where multiple peeks (using itertools) was already done. Was that in another branch or am I imagining things?

Comment (oberien) [reply src/generator/iter.rs:215]: It's required in the frontend for the pulldown-cmark `Parser`: https://github.com/oberien/heradoc/blob/b88acd8021bc751420cf33929a2ed5a65e620040/src/frontend/mod.rs#L39

Comment (197g) [reply src/generator/iter.rs:215]: So no conflict. Okay.

**Review by 197g — APPROVED** [review]

**Pull request merged**